src/Helper/CollectionHelper.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Helper;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Exception;
+
+/**
+ * Class CollectionHelper.
+ */
+class CollectionHelper
+{
+    private function __construct()
+    {
+    }
+
+    /**
+     * @throws Exception
+     */
+    public static function sort(Collection $collection, callable $comparator): ArrayCollection
```

Comment:
```
$sort = new Criteria(null, ['Order' => Criteria::ASC]);
return $this->collection->matching($sort);
```
Smthing like this could also do.

Reply:
@lutangar I wanted to be sure that I get sorted results even for data that has just been `setPinnedNotices` and not persisted yet …

Reply:
Does `matching` on `Criteria` work on in memory elements or executes a new query ? I’m unsure …

Reply:
I think it's in memory.

Reply:
mmmmm 

`The Criteria has a limited matching language that works both on the SQL and on the PHP collection level. This means you can use collection matching interchangeably, independent of in-memory or sql-backed collections.`

https://www.doctrine-project.org/projects/doctrine-orm/en/2.7/reference/working-with-associations.html

Nice ! I’m gonna try that !